table/block_based/filter_policy.cc

```diff
@@ -1790,8 +1785,8 @@ Status FilterPolicy::CreateFromString(
     } else {
       bloom_before_level = ParseInt(trim(value.substr(pos + 1)));
     }
-    double bloom_equivalent_bits_per_key =
-        ParseDouble(trim(value.substr(kRibbonName.size(), pos)));
```

Comment:
Is the previous parsing algorithm a bug which should have been `ParseDouble(
        trim(value.substr(kRibbonName.size(), pos - kRibbonName.size())));`?

Reply:
Yes but luckily it's just a code hygiene bug. If you go past the end with substr, it just gives you a string to the end. And ParseDouble seems to stop parsing and return what it has if it hits ':'.